www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php

```diff
@@ -66,73 +66,131 @@
-echo "UNKNOWN;" . $serviceStats["UNKNOWN_T"] . "s;" . $serviceStats["UNKNOWN_TP"] . "%;"
-    . $serviceStats["UNKNOWN_MP"] .  "%;" . $serviceStats["UNKNOWN_A"] . ";\n";
-echo _("SCHEDULED DOWNTIME") . ";" . $hostStats["MAINTENANCE_T"] . "s;"
+$serviceStats = getLogInDbForOneSVC(
+    $hostId,
+    $serviceId,
+    $start_date,
+    $end_date,
+    $reportingTimePeriod
+);
+
+echo "OK;"
+    . $serviceStats["OK_T"] . "s;"
+    . $serviceStats["OK_TP"] . "%;"
+    . $serviceStats["OK_MP"] . "%;"
+    . $serviceStats["OK_A"] . ";\n";
+
+echo "WARNING;"
+    . $serviceStats["WARNING_T"] . "s;"
+    . $serviceStats["WARNING_TP"] . "%;"
+    . $serviceStats["WARNING_MP"] . "%;"
+    . $serviceStats["WARNING_A"] . ";\n";
+
+echo "CRITICAL;"
+    . $serviceStats["CRITICAL_T"] . "s;"
+    . $serviceStats["CRITICAL_TP"] . "%;"
+    . $serviceStats["CRITICAL_MP"] . "%;"
+    . $serviceStats["CRITICAL_A"] . ";\n";
+
+echo "UNKNOWN;"
+    . $serviceStats["UNKNOWN_T"] . "s;"
+    . $serviceStats["UNKNOWN_TP"] . "%;"
+    . $serviceStats["UNKNOWN_MP"] .  "%;"
+    . $serviceStats["UNKNOWN_A"] . ";\n";
+
+echo _("SCHEDULED DOWNTIME") . ";"
+    . $hostStats["MAINTENANCE_T"] . "s;"
     . $hostStats["MAINTENANCE_TP"] . "%;;;\n";
-echo "UNDETERMINED;" . $serviceStats["UNDETERMINED_T"] . "s;"
+
+echo "UNDETERMINED;"
+    . $serviceStats["UNDETERMINED_T"] . "s;"
     . $serviceStats["UNDETERMINED_TP"] . "%;;;\n";
 echo "\n";
 echo "\n";
 
-/*
- * Getting evolution of service stats in time
- */
-echo _("Day").";"._("Duration").";"
-               ._("OK")." (s); "._("OK")." %; "._("OK")." Alert;"
-               ._("Warning")." (s); "._("Warning")." %;"._("Warning")." Alert;"
-               ._("Unknown")." (s); "._("Unknown")." %;"._("Unknown")." Alert;"
-               ._("Critical")." (s); "._("Critical")." %;"._("Critical")." Alert;"
-               ._("Day").";\n";
-
-$request = "SELECT  * FROM `log_archive_service` " .
-        "WHERE `host_id` = '".$host_id."' " .
-        "AND `service_id` = '".$service_id."' " .
-        "AND `date_start` >= '".$start_date."' " .
-        "AND `date_end` <= '".$end_date."' " .
-        "ORDER BY `date_start` DESC";
-$DBRESULT = $pearDBO->query($request);
-while ($row = $DBRESULT->fetchRow()) {
+// Getting evolution of service stats in time
+echo _("Day") . ";"
+    . _("Duration") . ";"
+    . _("OK") . " (s); "
+    . _("OK") . " %; "
+    . _("OK") . " Alert;"
+    . _("Warning") . " (s); "
+    . _("Warning") . " %;"
+    . _("Warning") . " Alert;"
+    . _("Unknown") . " (s); "
+    . _("Unknown") . " %;"
+    . _("Unknown") . " Alert;"
+    . _("Critical") . " (s); "
+    . _("Critical") . " %;"
+    . _("Critical") . " Alert;"
+    . _("Day") . ";\n";
+
+$dbResult = $pearDBO->query(
+    "SELECT  * FROM `log_archive_service` " .
+    "WHERE `host_id` = '" . $hostId . "' " .
+    "AND `service_id` = '" . $serviceId . "' " .
+    "AND `date_start` >= '" . $start_date . "' " .
```

Comment:
In this case, using a prepare statement is hightly recommended

Reply:
Refacto the whole code is out of my scope...  ;)